src/server/utils/bucket_diff.js

```diff
@@ -235,6 +250,41 @@ class BucketDiff {
         return ans;
     }
 
+    async get_first_bucket_keys_del(first_bucket_contents_left, first_bucket_delete_markers){
+        const first_bucket_delete_keys = [];
+        const field = 'IsLatest';
+        // Checking if there are any keys with latest delete marker in first bucket
+        Object.keys(first_bucket_contents_left).forEach(key => {
```

Comment:
In the case that you already holds the delete marker, all we need to do is check that it is latest, if it is then none of the versions are

Reply:
I need to check only for the latest delete marker and not for the version here, right?